nixos/modules/services/web-apps/libretranslate.nix

```diff
@@ -0,0 +1,203 @@
+        type = lib.types.package;
+        default = pkgs.libretranslate;
+        defaultText = lib.literalExpression "pkgs.libretranslate";
+        description = lib.mdDoc "LibreTranslate package to use.";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "User account under which libretranslate runs.";
+      };
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "Group account under which libretranslate runs.";
+      };
+      host = lib.mkOption {
+        description = lib.mdDoc "LibreTranslate host.";
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+      port = lib.mkOption {
+        description = lib.mdDoc "LibreTranslate port.";
+        type = lib.types.port;
+        default = 5000;
+      };
+      dataDir = lib.mkOption {
+        description = lib.mdDoc "The data directory.";
+        type = lib.types.path;
+        default = "/var/lib/libretranslate";
+        example = "/opt/libretranslate";
+      };
+      threads = lib.mkOption {
+        description = lib.mdDoc "Set number of threads.";
+        type = lib.types.nullOr lib.types.int;
+        default = null;
+        example = 8;
+      };
+      enableApiKeys = lib.mkOption {
+        description = lib.mdDoc "Enable API keys database.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      disableWebUI = lib.mkOption {
+        description = lib.mdDoc "Disable web ui.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      updateModels = lib.mkOption {
+        description = lib.mdDoc "Update language models at startup";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      domain = lib.mkOption {
+        description = lib.mdDoc "The domain serving your LibreTranslate instance. Required for configure nginx as a reverse proxy.";
+        type = lib.types.str;
+        default = "";
+        example = "libretranslate.example.com";
+      };
+      configureNginx = lib.mkOption {
+        description = lib.mdDoc "Configure nginx as a reverse proxy for LibreTranslate.";
+        type = lib.types.bool;
+        default = false;
+      };
+      extraArgs = lib.mkOption {
+        description = lib.mdDoc "Extra arguments passed to the LibreTranslate.";
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [
+          "--debug"
+          "--disable-files-translation"
+        ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
```

Comment:
```suggestion
```
the permissions should already be correct with the rule above

Reply:
This line automatically corrects permissions to folder if they have been changed.